applications: nrf_desktop: Document settings incompatibility (#3131)
nrfconnect/sdk-nrf: applications/nrf_desktop/README.rst

* There must be one partition for storing :ref:`zephyr:settings_api`.
* The bootloader, if enabled, will add additional partitions to the set.

> .. important::
>    Before updating the firmware, make sure that the data currently stored in settings partition is compatible with the new firmware.
>    In case of incompatibilities, the settings area must be erased before using the new firmware.

Reviewer: ```suggestion
.. important::
   Before updating the firmware, make sure that the data stored in the settings partition is compatible with the new firmware.
   If it is incompatible, erase the settings area before using the new firmware.
```
